### PR TITLE
Promotionreconciler: Stop retrying on requests that get a 404 from GH

### PR DESCRIPTION
**Pull request by alvaroaleman**
The 404 means that either:
* The repo/branch was deleted, so this request can not be reconciled
* We do not have access to this repo, which also means we can not
  reconcile it

The PR also removes the `ignoredGitHubOrganizations`, we instead try
everything and just stop retrying on the 404.

Fixes https://github.com/openshift/release/issues/9283

### CODE REVIEW

File: cmd/dptp-controller-manager/main.go [hunk 1]
```diff
@@ -43,7 +38,7 @@ func newOpts() (*options, error) {
 	flag.StringVar(&opts.configPath, "config-path", "", "Path to the prow config")
 	flag.StringVar(&opts.jobConfigPath, "job-config-path", "", "Path to the job config")
 	flag.StringVar(&opts.registryClusterKubeconfigPath, "registry-cluster-kubeconfig", "", "If set, this kubeconfig will be used to access registry-related resources like Images and ImageStreams. Defaults to the global kubeconfig")
-	flag.Var(&opts.promotionreconcilerOptions.IgnoredGitHubOrganizations, "promotionreconcilerOptions.ignored-github-organization", "GitHub organization to ignore in the imagestreamtagreconciler. Can be specified multiple times")
+	_ = flag.String("promotionreconcilerOptions.ignored-github-organization", "", "deprecated, doesn't do anything. Was used to ignore github organization. We instead attempt all repos and swallow 404 errors we get from github doing so.")
```
Comment (droslean) [inline cmd/dptp-controller-manager/main.go:41]: Let's just remove the whole thing.

Comment (alvaroaleman) [reply cmd/dptp-controller-manager/main.go:41]: That will send our deployment into crashlooping: https://github.com/openshift/release/blob/9e4b6fbfe9bc969a66e0b8b5a5a9844900021b44/clusters/app.ci/assets/dptp-controller-manager.yaml#L145

Comment (droslean) [reply cmd/dptp-controller-manager/main.go:41]: Can you open a follow-up PR in release repo to remove that as well?

Comment (alvaroaleman) [reply cmd/dptp-controller-manager/main.go:41]: Yup, that as the plan

**Pull request merged**